src/dfm/cleaning/quality_filter.py

```diff
@@ -0,0 +1,697 @@
+"""
+Danish implementation of quality filter and repetitious text filter described in [1]
+
+Authors:
+    Kenneth C. Enevoldsen
+    Kasper Junge
+    Malte Højmark-Bertelsen
+
+References:
+    [1] Rae, J. W., Borgeaud, S., Cai, T., Millican, K., Hoffmann, J., Song, F.,
+    Aslanides, J., Henderson, S., Ring, R., Young, S., Rutherford, E., Hennigan,
+    T., Menick, J., Cassirer, A., Powell, R., Driessche, G. van den, Hendricks,
+    L. A., Rauh, M., Huang, P.-S., … Irving, G. (2021).
+    Scaling Language Models: Methods, Analysis & Insights from Training Gopher.
+    https://arxiv.org/abs/2112.11446v2
+"""
+
+from typing import Iterable, Optional, Set, Tuple, List, Callable
+from collections import Counter, defaultdict
+from functools import partial
+
+import spacy
+from spacy.tokens import Doc
+
+
+def duplicate_chr_fraction_getter(doc: Doc, attr: str) -> float:
+    """Calculate the character fraction of duplicates based on a counter object"""
```

Comment:
Missing arguments and return types in docstring.

Reply:
added